proposals/Simple-programs.md

```diff
@@ -0,0 +1,169 @@
+# Simple programs
+
+* [x] Proposed
+* [x] Prototype: Started
+* [ ] Implementation: Not Started
+* [ ] Specification: Not Started
+
+## Summary
+[summary]: #summary
+
+Allow a sequence of *statements* to occur right before the *namespace_member_declaration*s of a *compilation_unit* (i.e. source file).
+
+The semantics are that if such a sequence of *statements* is present, the following type declaration, modulo the actual type name and the method name, would be emitted:
+
+``` c#
+static class Program
+{
+    static async Task Main()
```

Comment:
Can you return a Task<int>?
What about accessing args?

Reply:
> Can you return a Task<int>?

Not at the moment.

> What about accessing args?

At the moment the method has no args.


---
In reply to: [375636464](https://github.com/dotnet/csharplang/pull/3159#discussion_r375636464) [](ancestors = 375636464)